packages/ragbits-document-search/tests/unit/test_document_search.py

```diff
@@ -202,3 +216,256 @@ async def test_document_search_with_batched():
 
     assert len(await vectore_store.list()) == 12
     assert len(results) == 12
+
+
+@pytest.mark.asyncio
+async def test_document_search_ingest_from_uri_basic():
+    # Setup
+    with tempfile.TemporaryDirectory() as temp_dir:
+        test_file = Path(temp_dir) / "test.txt"
+        test_file.write_text("Test content")
+
+        document_search = DocumentSearch.from_config(CONFIG)
+
+        # Test ingesting from URI
+        await document_search.ingest(f"file://{test_file}")
+
+        # Verify
+        results = await document_search.search("Test content")
+        assert len(results) == 1
+        assert isinstance(results[0], TextElement)
+        assert isinstance(results[0].document_meta.source, LocalFileSource)
+        assert str(cast(LocalFileSource, results[0].document_meta.source).path) == str(test_file)
+        assert results[0].content == "Test content"
+
+
+@pytest.mark.asyncio
+async def test_document_search_ingest_from_uri_with_wildcard():
+    # Setup
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Create multiple test files
+        test_files = [
+            (Path(temp_dir) / "test1.txt", "First test content"),
+            (Path(temp_dir) / "test2.txt", "Second test content"),
+            (Path(temp_dir) / "other.txt", "Other content"),
+        ]
+        for path, content in test_files:
+            path.write_text(content)
+
+        document_search = DocumentSearch.from_config(CONFIG)
+
+        # Test ingesting from URI with wildcard
+        await document_search.ingest(f"file://{temp_dir}/test*.txt")
```

Comment:
It would be good to test other supported syntax options, for example `file://{temp_dir}/test?.txt`, `file://*.txt`, `file://**/test?.txt`